### PR TITLE
Add getters for some fields

### PR DESCRIPTION
**Pull request by lpy4105**
## Description

Partially Fix: https://github.com/Mbed-TLS/mbedtls/issues/6840

The getters are added for the convenience of application usage.

## Gatekeeper checklist

- [x] **changelog** provided
- [x] **backport** not required
- [x] **tests** provided

### CODE REVIEW

**Comment by yanesca** [comment]
@lpy4105 according to the new process only planned work that is scheduled for the current quarter should have 'high-priority' label. PRs with 'high-priority' label should be added to the unified board. This PR is labelled 'high-priority' but is not on the unified board. Does this PR belong to some stream of work we planned for this quarter?

**Comment by gilles-peskine-arm** [comment]
@yanesca This is from last quarter and I don't see any reason to abandon it.

**Comment by yanesca** [comment]
@gilles-peskine-arm thank you for confirming. I am not trying to challenge the necessity of this work or proposing to abandon it. There is an inconsistency between the number of items on the unified board and the items labelled as 'priority-high' and would like to figure out what is going on.

**Review by daverodgman — APPROVED** [review]

**Comment by daverodgman** [comment]
Please add tests and a Changelog entry.

**Review by daverodgman — CHANGES_REQUESTED** [review]
Please add tests & changelog

**Comment by lpy4105** [comment]
Adding tests results a conflict with `development` due to some changes in `tests/suites/test_suite_ssl.function`. So I rebased the PR to the latest `development`.

Changes:
- The parameter of `mbedtls_ssl_conf_get_endpoint` is changed to `const mbedtls_ssl_config *`
- Add tests for `mbedtls_ssl_conf_get_endpoint`, `mbedtls_ssl_get_hostname` and `mbedtls_ssl_cache_get_timeout`.
- Add changelog.

**Review by daverodgman — APPROVED** [review]

**Comment by daverodgman** [comment]
And please don't forget to update labels (@tom-daubney-arm remember to remove `needs-reviewer` when adding yourself as second reviewer)

File: tests/suites/test_suite_ssl.function
Comment (tom-daubney-arm) [inline tests/suites/test_suite_ssl.function:1158]: I don't really like the naming of the variables here. I know `hostname0` and `hostname1` existed before this PR but since we're here it might be nicer to rename them a bit for clarity. Maybe to distinguish between inputs and outputs so perhaps `hostname0` --> `input_hotstname0`, similar for `hostname1` and then perhaps `hostname` --> `ouput_hostname`.

This is just my preference and I accept others may not think this is needed so I won't request changes just for this, but consider changing them if you agree.

**Review by tom-daubney-arm — APPROVED** [review]
Overall LGTM.

A couple of minor points to address if you agree that they are beneficial changes but nothing that should hold up review. Thanks!

**Review by daverodgman — APPROVED** [review]

**Comment by daverodgman** [comment]
I think Tom's comments are sensible, but I don't think it's worth iterating the PR just for this.

**Pull request merged**
